app/models/deploy.rb

```diff
@@ -130,19 +130,43 @@ def self.expired
   end
 
   def buddy_name
-    user.id == buddy_id ? "bypassed" : buddy.try(:name)
+    user.id == buddy_id ? "bypassed" : buddy_with_deleted.try(:name)
   end
 
   def buddy_email
-    user.id == buddy_id ? "bypassed" : buddy.try(:email)
+    user.id == buddy_id ? "bypassed" : buddy_with_deleted.try(:email)
   end
 
   def url
     AppRoutes.url_helpers.project_deploy_url(project, self)
   end
 
+  def to_csv
+    [
+      id, project_with_deleted.name, summary, commit, job.status, updated_at,
+      start_time, user_with_deleted.try(:name), user_with_deleted.try(:email), buddy_name, buddy_email,
+      stage_with_deleted.production, stage_with_deleted.no_code_deployed, project_with_deleted.deleted_at
+    ]
+  end
+
   private
 
+  def buddy_with_deleted
+    buddy || (buddy_id && User.with_deleted { User.find(buddy_id) })
```

Comment:
need to cache all of these so they do not cause multiple queries ... also some tests for 'does no query when there is no id' ... 

```
return @foo if defined?(@foo)
@foo = bla || with_delete.find_by_id(xxx)
```

most likely also find_by_id so we don't blow up if something was hard-deleted ...